File: Search/Binary_Sarch/binary_search.cpp
#include <iostream>

using namespace std;

// Funtion returns the index of the number
// that we search for
int binary_search(int arr[], int number){
    int start = 0, end = number - 1;
    while (start <= end){
        int mid = (start + end)/2;
        if (arr[mid] == number){
            return mid;
        }if (arr[mid] > number){
            end = mid - 1;
        }else{
            start = mid + 1;
        }
    }
    
}

File: Search/Binary_Sarch/binary_search.py

def binary_search(arr, num):
    start = 0, end = len(arr) - 1
    while start <= end:
        mid = (star + end)/2
        if arr[mid] == num:
            return mid
        elif arr[mid] > num:
            end = mid - 1
        else:
            start = mid + 1